chromium_src/chrome/browser/resources/tab_search/auto_tab_groups/auto_tab_groups_page.ts

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import 'chrome://resources/cr_elements/cr_button/cr_button.js'
+import 'chrome://resources/cr_elements/cr_input/cr_input.js'
+
+import {loadTimeData} from 'chrome://resources/js/load_time_data.js'
+import {CrLitElement} from 'chrome://resources/lit/v3_0/lit.rollup.js'
+
+import {BraveTabSearchApiProxy, TabSearchApiProxyImpl} from '../tab_search_api_proxy.js'
+import type {TabOrganizationSession} from '../tab_search.mojom-webui.js'
+
+import {getHtml} from './auto_tab_groups_page.html.js'
+import {getCss} from './auto_tab_groups_page.css.js'
+
+export class AutoTabGroupsPageElement extends CrLitElement {
+  static get is() {
+    return 'auto-tab-groups-page'
+  }
+
+  protected topics_: string[] = []
+  protected topic: string = ''
+  protected undoTopic_: string = ''
+
+  private apiProxy_: BraveTabSearchApiProxy =
+      TabSearchApiProxyImpl.getInstance() as BraveTabSearchApiProxy
+
+  private visibilityChangedListener_: () => void
+  private elementVisibilityChangedListener_: IntersectionObserver
+  private isElementVisible_: boolean = false
+
+  static override get properties() {
+    return {
+      availableHeight: {type: Number},
+      showBackButton: {type: Boolean},
+      topic: {type: String},
+      topics_: {type: Array},
+      undoTopic_: {type: String},
+      isLoadingTopics: {type: Boolean},
+      errorMessage: {type: String},
+      needsPremium: {type: Boolean},
+    }
+  }
+
+  availableHeight: number = 0
+  showBackButton: boolean = false
+  isLoadingTopics: boolean = false
+  errorMessage: string = ''
+  needsPremium: boolean = false
+
+  static override get styles() {
+    return getCss()
+  }
+
+  constructor() {
+    super()
+
+    this.visibilityChangedListener_ = () => {
+      this.maybeUpdateSuggestedTopics_()
+    }
+
+    this.elementVisibilityChangedListener_ =
+        new IntersectionObserver((entries, _observer) => {
+          entries.forEach(entry => {
+            this.onElementVisibilityChanged_(entry.intersectionRatio > 0)
+          })
+        }, {root: document.documentElement})
+
+  }
+
+  protected onTopicInputChanged_(e: any) {
+    this.topic = e.value
+  }
+
+  protected onFocusTabsClicked_() {
+    this.getFocusTabs_(this.topic)
+  }
+
+  protected onUndoClicked_() {
+    this.apiProxy_.undoFocusTabs().then(() => {
+      this.undoTopic_ = ''
+    })
+  }
+
+  private getFocusTabs_(topic: string) {
+    this.undoTopic_ = ''
```

Comment:
optional: I wonder if `undoTopic_` should be `string | null` instead of using the empty string as falsey